Makefile

```diff
@@ -104,14 +106,33 @@ clean: ## Remove all generated files
 	rm -f bazel-*
 
 .PHONY: docker-build
-docker-build: generate fmt vet manifests ## Build the docker image
-	docker build . -t ${IMG}
+docker-build: generate fmt vet manifests ## Build the docker image for controller-manager
+	docker build . -t ${CONTROLLER_IMG}
 	@echo "updating kustomize image patch file for manager resource"
-	sed -i.tmp -e 's@image: .*@image: '"${IMG}"'@' ./config/default/manager_image_patch.yaml
+	sed -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}"'@' ./config/default//manager_image_patch.yaml
 
 .PHONY: docker-push
-docker-push: ## Push the docker image
-	docker push ${IMG}
+docker-push: docker-build ## Push the docker image
+    # Log into the registry with a service account file. GCR_KEY_FILE contains the content and not the file name.
+	@echo "logging into gcr.io registry with key file"
+	@echo $$GCR_KEY_FILE | docker login -u _json_key --password-stdin gcr.io
```

Comment:
Why is docker login needed?

Reply:
@vincepri  i removed the login part now since we are leverage KIND side loader, there is no need to login and publish container to public registry now.